pageserver/src/tenant/timeline.rs

```diff
@@ -3765,15 +3773,11 @@ impl Timeline {
     /// Request the flush loop to write out all frozen layers up to `to_lsn` as Delta L0 files to disk.
     /// The caller is responsible for the freezing, e.g., [`Self::freeze_inmem_layer`].
     ///
-    /// `last_record_lsn` may be higher than the highest LSN of a frozen layer: if this is the case,
-    /// it means no data will be written between the top of the highest frozen layer and to_lsn,
-    /// e.g. because this tenant shard has ingested up to to_lsn and not written any data locally for that part of the WAL.
-    async fn flush_frozen_layers_and_wait(
-        &self,
-        last_record_lsn: Lsn,
-    ) -> Result<(), FlushLayerError> {
-        let mut rx = self.layer_flush_done_tx.subscribe();
-
+    /// `last_record_lsn` may be higher than the highest LSN of a frozen layer: if this is the
```

Comment:
nit: do you mean `at_lsn` now?

Reply:
yes this needs fixing.

Reply:
Previously, `flush_frozen_layers_and_wait` was always called with the last record lsn, hence the param name (i suppose). Not sure I answered your question ...

Reply:
I was wondering about the: https://github.com/neondatabase/neon/blob/b52e31c1a42e186d578975cce632bf244c5f2957/pageserver/src/tenant/timeline.rs#L3771-L3773

Because I can see `LayerMap::open_layer_at` but I don't see any writes to that variable except from `LayerManager::try_freeze_in_memory_layer` (modified in this PR).

Hmm... It makes me think we do not have a test for this.

Reply:
Nevermind I always forget it's at the end of flush loop:

https://github.com/neondatabase/neon/blob/a8ca7a1a1d88c6cff476eaf55c9f38c46dbfc645/pageserver/src/tenant/timeline.rs#L3746-L3759